VeganVirus/VeganVirus/SoundAction.cpp

```diff
@@ -2,42 +2,46 @@
 
 SoundAction::SoundAction(double req) : Action(req, MUSIC_ICON_PATH)
 {
-    playListTime[0] = 62;
+    _timer = 0;
 }
 
 void SoundAction::start()
 {
-    if (!canActivate())
+    _timer = SOUND_DUARTION;
+}
+
+void SoundAction::update(double dt)
+{
+    std::string song = "";
+    _timer -= dt;
+    if (_timer <= 0 || !canActivate())
```

Comment:
This condition is a bit problematic. It's also unnecessary if you move the rest of this method to `start` like I suggested.

I suggest removing everything in this method instead of the line `_timer -= dt;`